web/jobs/jobs.go

```diff
@@ -612,3 +618,15 @@ func wrapJobsError(err error) error {
 	}
 	return err
 }
+
+// isReservedWorker returns true if the worker should only by used by the
+// stack, and the clients must not push jobs for it.
+func isReservedWorker(worker string) bool {
+	switch worker {
+	case "migrations", "thumbnail", "trash-files", "updates",
+		"share-track", "share-replicate", "share-upload", "":
```

Comment:
I think it would be better to mark the worker as "reserved" in the definition/code of the worker itself (in a boolean property by example) rather than in a separate hardcoded blacklist.